featuretools/entityset/entityset.py

```diff
@@ -163,76 +155,84 @@ def is_metadata(self):
-            } for r in self.relationships],
-            'entity_dict': {eid: {
-                'index': e.index,
-                'time_index': e.time_index,
-                'secondary_time_index': e.secondary_time_index,
-                'variables': {
-                    v.id: v.create_metadata_dict()
-                    for v in e.variables
-                },
-                'has_last_time_index': e.last_time_index is not None
-            } for eid, e in self.entity_dict.items()},
-        }
+    def create_data_description(self):
+        '''Serialize entityset to data description.
+
+        Args:
+            es (EntitySet) : Instance of :class:`.EntitySet`.
+
+        Returns:
+            description (dict) : Description of :class:`.EntitySet`.
+        '''
+        entities = {entity.id: serialization.to_entity_description(entity) for entity in self.entities}
+        relationships = [serialization.to_relationship_description(relationship) for relationship in self.relationships]
+        return {'id': self.id, 'entities': entities, 'relationships': relationships}
 
     @classmethod
-    def from_metadata(cls, metadata, data_root=None):
-        es = EntitySet(metadata['id'])
-        set_last_time_indexes = False
-        for eid, entity in metadata['entity_dict'].items():
-            df, variable_types = cls._load_dummy_entity_data_and_variable_types(entity)
-            if data_root is not None:
-                df = load_entity_data(entity, data_root)
-            es.entity_from_dataframe(eid,
-                                     df,
-                                     index=entity['index'],
-                                     time_index=entity['time_index'],
-                                     secondary_time_index=entity['secondary_time_index'],
-                                     variable_types=variable_types)
-            if entity['has_last_time_index']:
-                set_last_time_indexes = True
-            for vid, v in entity['variables'].items():
-                if v['interesting_values'] and len(v['interesting_values']):
-                    es[eid][vid].interesting_values = v['interesting_values']
-        for rel in metadata['relationships']:
-            es.add_relationship(Relationship(
-                es[rel['parent_entity']][rel['parent_variable']],
-                es[rel['child_entity']][rel['child_variable']],
-            ))
-        if set_last_time_indexes:
-            es.add_last_time_indexes()
-        return es
+    def from_data_description(cls, description, **kwargs):
+        '''Deserialize entityset from data description.
+
+        Args:
+            description (dict) : Description of :class:`.EntitySet`.
+            kwargs (keywords): Additional keyword arguments to pass as keywords arguments to the underlying deserialization method.
+
+        Returns:
+            entityset (EntitySet) : Instance of :class:`.EntitySet`.
+        '''
+        path = description.get('root')
```

Comment:
can we add a comment here or and whereever it makes sense how sometimes Path will be None and when it is None no data is loaded

Reply:
Change applied in [this commit.](https://github.com/Featuretools/featuretools/pull/361/commits/b23325f9bd271f07c3cc174e627db41089faedef)